### PR TITLE
"activespecial" IPC event

### PR DESCRIPTION
**Pull request by MightyPlaza**
Adds "activespecial" IPC event
Also some minor bug fixes

no issues found
probably ready for merge

### CODE REVIEW

File: src/helpers/Monitor.cpp [hunk 1]
```diff
@@ -572,8 +572,10 @@ void CMonitor::setSpecialWorkspace(CWorkspace* const pWorkspace) {
 
     if (!pWorkspace) {
         // remove special if exists
-        if (const auto EXISTINGSPECIAL = g_pCompositor->getWorkspaceByID(specialWorkspaceID); EXISTINGSPECIAL)
+        if (const auto EXISTINGSPECIAL = g_pCompositor->getWorkspaceByID(specialWorkspaceID); EXISTINGSPECIAL) {
             EXISTINGSPECIAL->startAnim(false, false);
+            g_pEventManager->postEvent(SHyprIPCEvent{"activespecial", "none," + szName});
```
Comment (vaxerski) [inline src/helpers/Monitor.cpp:577]: I'd rather have it empty if closed than `none`

Comment (MightyPlaza) [reply src/helpers/Monitor.cpp]: done

Comment (MightyPlaza) [reply src/helpers/Monitor.cpp]: Now I am not sure if you meant "empty" or ""

Comment (vaxerski) [reply src/helpers/Monitor.cpp]: I meant `""`

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**